File: ecommerce-order/src/main/java/com/ecommerce/order/entity/Invoice.java
package com.ecommerce.order.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.ecommerce.common.mybatis.entity.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("t_invoice")
public class Invoice extends BaseEntity {

    private Long orderId;
    private Long userId;
    /** 1=个人, 2=企业 */
    private Integer type;
    /** 发票抬头 */
    private String title;
    /** 税号(企业必填) */
    private String taxNo;
    private BigDecimal amount;
    /** 接收邮箱 */
    private String email;
    /** 联系电话 */
    private String phone;
    /** 0=待开票, 1=已开票, 2=已拒绝 */
    private Integer status;
    /** 发票号码 */
    private String invoiceNo;
    /** 电子发票PDF地址 */
    private String invoiceUrl;
    private String remark;
    private String rejectReason;
    private LocalDateTime applyTime;
    private LocalDateTime issueTime;
}

File: ecommerce-frontend/packages/shared/src/api/invoice.ts
import { get, post, put } from './request'
import type { PageResult } from '../types/index'

export interface InvoiceDTO {
  id: number
  orderId: number
  orderNo?: string
  userId: number
  type: number
  title: string
  taxNo: string
  amount: number
  email: string
  phone: string
  status: number
  statusName: string
  invoiceNo: string
  invoiceUrl: string
  remark: string
  rejectReason: string
  applyTime: string
  issueTime: string
  createTime: string
}

export interface InvoiceApplyRequest {
  orderId: number
  type: number
  title: string
  taxNo?: string
  email?: string
  phone?: string
  remark?: string
}

export const invoiceApi = {
  apply: (data: InvoiceApplyRequest) =>
    post<void>('/order/order/invoice/apply', data),

  getMyInvoices: (params: { pageNum?: number; pageSize?: number }) =>
    get<PageResult<InvoiceDTO>>('/order/order/invoice/my', params),

  getInvoice: (id: number) =>
    get<InvoiceDTO>(`/order/order/invoice/${id}`),

  listAdmin: (params: { pageNum?: number; pageSize?: number; status?: number }) =>
    get<PageResult<InvoiceDTO>>('/order/order/admin/invoice/list', params),

  issue: (id: number, data: { invoiceNo: string; invoiceUrl: string }) =>
    put<void>(`/order/order/admin/invoice/${id}/issue`, data),

  reject: (id: number, data: { reason: string }) =>
    put<void>(`/order/order/admin/invoice/${id}/reject`, data),
}
